File: src/tutorial_cpp_pkg/src/service_server.cpp
#include "rclcpp/rclcpp.hpp"
#include "tutorial_cpp_pkg/srv/test_service.hpp"

class MyNode: public rclcpp::Node
{
public:
    MyNode() : Node("CPP_Service_Server") {
        RCLCPP_INFO(this->get_logger(), "Cpp Service Server");
        server_ = this->create_service<tutorial_cpp_pkg::srv::TestService>("cpp_tutorial_service",
            std::bind(&MyNode::callbackService, this, std::placeholders::_1, std::placeholders::_2)
            );
        RCLCPP_INFO(this->get_logger(), "Start service...");
    }

private:
    void callbackService(const tutorial_cpp_pkg::srv::TestService::Request::SharedPtr request,
                         const tutorial_cpp_pkg::srv::TestService::Response::SharedPtr response) {
        response->sum = 0.0;
        for(double data: request->datas) {
            response->sum += data;
        }
        RCLCPP_INFO(this->get_logger(), "Result: %lf", response->sum);
    }

private:
    rclcpp::Service<tutorial_cpp_pkg::srv::TestService>::SharedPtr server_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<MyNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/tutorial_py_pkg/tutorial_py_pkg/service_server.py
import rclpy
from rclpy.node import Node
from tutorial_cpp_pkg.srv import TestService

class MyNode(Node):
    def __init__(self):
        super().__init__("Python_Service_Server")
        self.server_ = self.create_service(TestService, "py_tutorial_service", self.callback_service)
        self.get_logger().info("Start service...")
        
    def callback_service(self, req, res):
        res.sum = 0.0
        for data in req.datas:
            res.sum += data
        self.get_logger().info("Service result: " + str(res.sum))
        return res

def main(args=None):
    rclpy.init(args=args)
    node = MyNode()
    rclpy.spin(node)
    rclpy.shutdown()
    
if __name__ == "__main__":
    main()